aws/resource_aws_docdb_cluster_instance_test.go

```diff
@@ -286,84 +291,107 @@ func testAccCheckAWSDocDBClusterInstanceExists(n string, v *docdb.DBInstance) re
+    data.aws_availability_zones.available.names[2]
+  ]
   master_username     = "foo"
   master_password     = "mustbeeightcharaters"
   skip_final_snapshot = true
 }
 
+data "aws_docdb_orderable_db_instance" "test" {
+  engine                     = "docdb"
+  preferred_instance_classes = ["db.t3.medium", "db.r4.large", "db.r5.large", "db.r5.xlarge"]
+}
+
 resource "aws_docdb_cluster_instance" "cluster_instances" {
-  identifier                 = "tf-cluster-instance-%d"
+  identifier                 = %[1]q
   cluster_identifier         = aws_docdb_cluster.default.id
-  instance_class             = "db.r4.large"
+  instance_class             = data.aws_docdb_orderable_db_instance.test.instance_class
   auto_minor_version_upgrade = false
   promotion_tier             = "3"
 }
-`, n, n)
-}
-
-func testAccAWSDocDBClusterInstanceConfig_az(n int) string {
-	return fmt.Sprintf(`
-data "aws_availability_zones" "available" {
-  state = "available"
-
-  filter {
-    name   = "opt-in-status"
-    values = ["opt-in-not-required"]
-  }
+`, rName))
 }
 
+func testAccAWSDocDBClusterInstanceConfig_az(rName string) string {
+	return composeConfig(testAccAvailableAZsNoOptInConfig(), fmt.Sprintf(`
 resource "aws_docdb_cluster" "default" {
-  cluster_identifier  = "tf-docdb-cluster-test-%d"
-  availability_zones  = [data.aws_availability_zones.available.names[0], data.aws_availability_zones.available.names[1], data.aws_availability_zones.available.names[2]]
+  cluster_identifier  = %[1]q
+  availability_zones  = [
+    data.aws_availability_zones.available.names[0],
+    data.aws_availability_zones.available.names[1],
+    data.aws_availability_zones.available.names[2]
+  ]
   master_username     = "foo"
   master_password     = "mustbeeightcharaters"
   skip_final_snapshot = true
 }
 
+data "aws_docdb_orderable_db_instance" "test" {
+  engine                     = "docdb"
+  preferred_instance_classes = ["db.t3.medium", "db.r4.large", "db.r5.large", "db.r5.xlarge"]
+}
+
 resource "aws_docdb_cluster_instance" "cluster_instances" {
-  identifier         = "tf-cluster-instance-%d"
+  identifier         = %[1]q
   cluster_identifier = aws_docdb_cluster.default.id
-  instance_class     = "db.r4.large"
+  instance_class     = data.aws_docdb_orderable_db_instance.test.instance_class
   promotion_tier     = "3"
   availability_zone  = data.aws_availability_zones.available.names[0]
 }
-`, n, n)
+`, rName))
+}
+
+func testAccAWSDocDBClusterInstanceConfig_namePrefix(rName string) string {
+	return composeConfig(testAccAvailableAZsNoOptInConfig(), fmt.Sprintf(`
+data "aws_docdb_orderable_db_instance" "test" {
+  engine                     = "docdb"
+  preferred_instance_classes = ["db.t3.medium", "db.r4.large", "db.r5.large", "db.r5.xlarge"]
 }
 
-func testAccAWSDocDBClusterInstanceConfig_namePrefix(n int) string {
-	return fmt.Sprintf(`
 resource "aws_docdb_cluster_instance" "test" {
-  identifier_prefix  = "tf-cluster-instance-"
+  identifier_prefix  = "tf-acc-docdb-"
```

Comment:
Looks like just this small little thing:

```
=== CONT  TestAccAWSDocDBClusterInstance_namePrefix
TestAccAWSDocDBClusterInstance_namePrefix: resource_aws_docdb_cluster_instance_test.go:104: Check 4/4 error: aws_docdb_cluster_instance.test: Attribute 'identifier' didn't match "^tf-acc-test-", got "tf-acc-docdb-20200910175222397800000001"
--- FAIL: TestAccAWSDocDBClusterInstance_namePrefix (472.19s)
```

Reply:
Fixed.